scirpy/_tools/_diversity.py

```diff
@@ -65,8 +91,24 @@ def _shannon_entropy(freq):
     diversity = dict()
     for k in sorted(ir_obs[groupby].unique()):
         tmp_counts = clono_counts.loc[clono_counts[groupby] == k, "count"].values
-        tmp_freqs = tmp_counts / np.sum(tmp_counts)
-        diversity[k] = _shannon_entropy(tmp_freqs)
+        
+        if method in ["normalized shannon entropy", "D50"]:
+            # calculate frequencies for these two metrics
+            tmp_freqs = tmp_counts / np.sum(tmp_counts)
+            np.testing.assert_almost_equal(np.sum(tmp_freqs), 1)
+
+            if method == "normalized shannon entropy":
+                diversity[k] = _shannon_entropy(tmp_freqs)
+            else:
+                diversity[k] = _dxx(tmp_freqs, perc=0.5)
+        
+        elif method in skbio.diversity.get_alpha_diversity_metrics():
```

Comment:
I'm a bit wary of adding too many dependencies... Could you please make skbio an *optional dependency*? See an example [here](https://github.com/icbi-lab/scirpy/blob/507c943e50a2d2ba5b0011575030db85dd5b316b/scirpy/util/graph/_component_layout.py#L87-L94).